File: NirvanaSMS.Api.Java/src/main/java/org/nirvanasms/api/core/entity/ManyToManyMessage.java
package org.nirvanasms.api.core.entity;

import java.util.List;

import org.nirvanasms.api.core.base.MessageBase;
import org.nirvanasms.api.core.utility.ManyToManyMessageBuilder;
import org.nirvanasms.api.core.utility.OneToManyMessageBuilder;

public class ManyToManyMessage extends MessageBase {
	private List<PhoneAndMessagePair> phonesAndMessages;

	public ManyToManyMessage() {

	}

	public ManyToManyMessage(ManyToManyMessageBuilder builder) {
		super.setHeader(builder.getHeader());
		this.phonesAndMessages = builder.getPhonesAndMessages();
		super.setValidityTime(builder.getValidityTime());
		super.setSendDateTime(builder.getSendDateTime());
	}

	public List<PhoneAndMessagePair> getPhonesAndMessages() {
		return phonesAndMessages;
	}

	public void setPhonesAndMessages(List<PhoneAndMessagePair> phonesAndMessages) {
		this.phonesAndMessages = phonesAndMessages;
	}

	public ManyToManyMessageBuilder builder(){
		return new ManyToManyMessageBuilder();
	}
}

File: NirvanaSMS.Api.CSharp/LibraryProject/Core/Entity/ManyToManyMessage.cs
using NirvanaSMS.Api.Core.Base;
using NirvanaSMS.Api.Core.Utility;
using System.Collections.Generic;

namespace NirvanaSMS.Api.Core.Entity
{
    public class ManyToManyMessage : MessageBase
    {
        public List<PhoneAndMessagePair> PhonesAndMessages { get; set; }
        public ManyToManyMessageBuilder Builder { get; set; }

        public ManyToManyMessage()
        {
            Builder = new ManyToManyMessageBuilder();
        }

        public ManyToManyMessage(ManyToManyMessageBuilder builder)
        {
            Header = builder.Header;
            PhonesAndMessages = builder.PhonesAndMessages;
            ValidityTime = builder.ValidityTime;
            SendDateTime = builder.SendDateTime;
        }
    }
}
